### PR TITLE
Numerical column stats update

### PR DESCRIPTION
**Pull request by atl1502**

### CODE REVIEW

File: dataprofiler/profilers/numerical_column_stats.py [hunk 1]
```diff
@@ -1741,8 +1753,26 @@ def _update_helper(self, df_series_clean: pd.Series, profile: dict) -> None:
         :type profile: dict
         :return: None
         """
-        if df_series_clean.empty:
-            return
+        self._greater_than_64_bit = (
+            not df_series_clean.empty
+            and df_series_clean.apply(pd.to_numeric, errors="coerce").dtype == "O"
+        )
+        if self._greater_than_64_bit:
```
Comment (ksneab7) [inline dataprofiler/profilers/numerical_column_stats.py:1760]: Good design choice here I like it

**Review by ksneab7 — APPROVED** [review]

File: dataprofiler/profilers/numerical_column_stats.py [hunk 2]
```diff
@@ -1741,8 +1753,26 @@ def _update_helper(self, df_series_clean: pd.Series, profile: dict) -> None:
         :type profile: dict
         :return: None
         """
-        if df_series_clean.empty:
-            return
+        self._greater_than_64_bit = (
```
Comment (micdavis) [inline dataprofiler/profilers/numerical_column_stats.py:1756]: what does `_greater_than_64_bit` mean? If it's true, what should that tell me? If it's false?

Comment (micdavis) [reply dataprofiler/profilers/numerical_column_stats.py:1756]: Oh does `_greater_than_64_bit` mean that it's been converted to numpy if it's true, and it is still pandas if it's false?

File: dataprofiler/profilers/numerical_column_stats.py [hunk 3]
```diff
@@ -1963,14 +2022,16 @@ def _get_num_zeros(
         :type subset_properties: dict
         :return: None
         """
+        if not self._greater_than_64_bit:
+            df_series = pl.from_pandas(df_series)
```
Comment (micdavis) [inline dataprofiler/profilers/numerical_column_stats.py:2026]: Is this supposed to be converting `pd.Series` to `np.ndarray`? I read `if not self._greater_than_64_bit` as "if this isn't a ndarray". Correct me if I am not following the logic correctly.

**Review by micdavis — DISMISSED** [review]

**Review by micdavis — APPROVED** [review]

File: dataprofiler/profilers/numerical_column_stats.py [hunk 4]
```diff
@@ -1765,81 +1800,111 @@ def _update_helper(self, df_series_clean: pd.Series, profile: dict) -> None:
...
-        min_value = df_series.min()
-        self.min = min_value if not self.min else min(self.min, min_value)
+        if self._greater_than_64_bit:
+            min_value = np.min(df_series)
+            self.min = min_value if not self.min else min(self.min, min_value)
+        else:
+            df_series = pl.from_pandas(df_series)
+            min_value = df_series.min()
+            if self.min is not None:
+                min_value = type(self.min)(min_value)
```
[the remark this answers is not in the document]
Comment (atl1502) [reply dataprofiler/profilers/numerical_column_stats.py:1814]: This is casting min_value to the current type of self.min

File: dataprofiler/profilers/numerical_column_stats.py [hunk 5]
```diff
@@ -1765,81 +1800,111 @@ def _update_helper(self, df_series_clean: pd.Series, profile: dict) -> None:
...
+                batch_biased_variance,
+                batch_mean,
+            )
         )
 
     @BaseColumnProfiler._timeit(name="skewness")
     def _get_skewness(
         self,
-        df_series: pd.Series,
+        df_series: pd.Series | np.ndarray,
```
[the remark this answers is not in the document]
Comment (atl1502) [reply dataprofiler/profilers/numerical_column_stats.py:1907]: yup

File: dataprofiler/profilers/numerical_column_stats.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (atl1502) [reply dataprofiler/profilers/numerical_column_stats.py:1814]: This is not needed actually updating it to remove this line

**Review by taylorfturner — APPROVED** [review]

**Review by micdavis — APPROVED** [review]

**Pull request merged**